packages/state-manager/src/sagas/messages/verifyMessage/verifyMessages.saga.test.ts

```diff
@@ -0,0 +1,300 @@
+import { setupCrypto, pubKeyFromCsr } from '@quiet/identity'
+import { type Store } from '../../store.types'
+import { getFactory } from '../../..'
+import { prepareStore, reducers } from '../../../utils/tests/prepareStore'
+import { combineReducers } from '@reduxjs/toolkit'
+import { expectSaga } from 'redux-saga-test-plan'
+import { type communitiesActions } from '../../communities/communities.slice'
+import { type identityActions } from '../../identity/identity.slice'
+import { type FactoryGirl } from 'factory-girl'
+import { generateChannelId, userCreatedChannelMessage, userJoinedMessage, verifyUserInfoMessage } from '@quiet/common'
+import { publicChannelsActions } from '../../publicChannels/publicChannels.slice'
+import { DateTime } from 'luxon'
+import {
+  type Community,
+  type Identity,
+  MessageType,
+  type PublicChannel,
+  ChannelMessage,
+  IncomingMessages,
+} from '@quiet/types'
+import { verifyMessagesSaga } from './verifyMessages.saga'
+import { messagesActions } from '../messages.slice'
+import { messagesSelectors } from '../messages.selectors'
+import { usersSelectors } from '../../users/users.selectors'
+
+describe('verifyMessage saga test', () => {
+  let store: Store
+  let factory: FactoryGirl
+
+  let community: Community
+  let alice: Identity
+  let bob: Identity
+
+  let generalChannel: PublicChannel
+  let sportChannel: PublicChannel
+
+  beforeAll(async () => {
+    setupCrypto()
+
+    store = prepareStore().store
+
+    factory = await getFactory(store)
+
+    community = await factory.create<ReturnType<typeof communitiesActions.addNewCommunity>['payload']>('Community')
+
+    alice = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>('Identity', {
+      id: community.id,
+      nickname: 'alice',
+    })
+
+    bob = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>('Identity', {
+      id: community.id,
+      nickname: 'bob',
+    })
+
+    generalChannel = (
+      await factory.create<ReturnType<typeof publicChannelsActions.addChannel>['payload']>('PublicChannel', {
+        channel: {
+          name: 'general',
+          description: 'Welcome to #general',
+          timestamp: DateTime.utc().valueOf(),
+          owner: alice.nickname,
+          id: generateChannelId('general'),
+        },
+      })
+    ).channel
+
+    sportChannel = (
+      await factory.create<ReturnType<typeof publicChannelsActions.addChannel>['payload']>('PublicChannel', {
+        channel: {
+          name: 'sport',
+          description: 'Welcome to #sport',
+          timestamp: DateTime.utc().valueOf(),
+          owner: alice.nickname,
+          id: generateChannelId('sport'),
+        },
+      })
+    ).channel
+  })
+
+  it('verify standard message ', async () => {
+    if (!alice.userCsr?.userCsr) throw Error('no Alice userCsr')
```

Comment:
Funny thing to do in test. Verifies mocked data validity. What does it result with when running with no csr generated? Will it mark the test as failed?